src/components/Entry.js

```diff
@@ -10,31 +10,44 @@ class Entry extends Component {
     this.handleChangeStatus = this.handleChangeStatus.bind(this);
   }
 
-  handleChangeStatus(e) {
-    const status = e.target.value;
+  handleChangeStatus(status) {
     this.props.changeEntryStatus(this.props.index, status);
   }
 
+  renderActionButtons() {
+    return dom.td(
+      { className: 'entry-actions col-md-2' },
+      this.props.status === constants.OPEN ? dom.button(
+        {
+          className: 'btn btn-primary btn-block delivered',
+          onClick: this.handleChangeStatus.bind(null, constants.DELIVERED)
+        },
+        'Delivered'
+      ) : dom.button(
+      {
+        className: 'btn btn-primary btn-block open',
+        onClick: this.handleChangeStatus.bind(null, constants.OPEN)
+      },
+      'Reopen'
+      ),
+      dom.button(
+        {
+          className: 'btn btn-warning btn-block canceled',
+          onClick: this.handleChangeStatus.bind(null, constants.CANCELED)
+        },
+        'Cancel'
+      )
+    );
+  }
+
   render() {
     return dom.tr(
       { className: 'order-entry' },
       dom.td({ className: 'entry-name' }, dom.h2(null, this.props.name)),
       dom.td({ className: 'entry-comment' }, dom.h2(null, dom.small(null, this.props.comment))),
+      this.props.ofOpenOrder ? dom.td({ className: 'entry-status col-md-2' }, dom.h2(null, dom.small(null, this.props.status))) : null,
```

Comment:
this is kind of whack - where it's checking this.props.ofOpenOrder twice. tried having renderActionButtons to return two tds (for status, and action buttons) when true, else return one td (for price) doesn't work.